src/AWS.Deploy.CLI/Commands/ServerModeCommand.cs

```diff
@@ -35,7 +33,58 @@ public async Task ExecuteAsync(CancellationToken cancellationToken = default(Can
 
             var host = builder.Build();
 
-            await host.RunAsync(cancellationToken); 
+            if (_parentPid == null)
+            {
+                await host.RunAsync(cancellationToken);
+            }
+            else
+            {
+                var monitorTask = WaitOnParentPid(cancellationToken);
+                var webTask = host.RunAsync(cancellationToken);
+
+                //This call will wait until one of the tasks completes.
+                //The monitor task will complete if a parent pid is not found.
+                await Task.WhenAny(monitorTask, webTask);
+
+                //The monitor task is expected to complete only when a parent pid is not found.
+                if (monitorTask.IsCompleted && monitorTask.Result?.ProcessNotFound == true)
+                {
+                    throw new ParentProcessNotFoundException("The parent process is no longer running.");
```

Comment:
I don't think we have to throw an exception causing a non-zero exit code. I think we should just log that the parent process exited so we are shutting down server mode.

Reply:
Fixed.